src/metabase/lib/schema/ref.cljc

```diff
@@ -0,0 +1,58 @@
+(ns metabase.lib.schema.ref
+  "Malli schema for a Field, aggregation, or expression reference (etc.)"
+  (:require
+   [metabase.lib.schema.common :as common]
+   [metabase.lib.schema.id :as id]
+   [metabase.lib.schema.temporal-bucketing :as temporal-bucketing]
+   [metabase.types]
+   [metabase.util.malli.registry :as mr]))
+
+(comment metabase.types/keep-me)
+
+(mr/def ::field.options
+  [:and
+   ::common/options
+   [:map
+    [:temporal-unit {:optional true} ::temporal-bucketing/unit]]])
```

Comment:
I'm not sure if this works like we'd hope, merging the map keys together. `malli.util/union` definitely does this, though, so let's use that?

Reply:
I tried that and it didn't seem to work with schemas that it had to resolve thru the registry, but maybe I did it wrong